frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/SignUpHandler.java

```diff
@@ -107,6 +108,13 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
 
         attachSessionIdToLogs(userContext.getSession());
 
+        var optionalAuthSessionItem =
+                authSessionService.getSessionFromRequestHeaders(input.getHeaders());
+        if (optionalAuthSessionItem.isEmpty()) {
+            return generateApiGatewayProxyErrorResponse(400, ErrorResponse.ERROR_1000);
```

Comment:
Is this defo a 400? What if the request is valid but the session can't be found for another reason? 
Maybe return 400 for missing Session-Id and 500 for session not found?

Reply:
Good points, I agree with you now 😄